src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1546,6 +1594,62 @@ protected override void HandleThreadsRequestAsync(IRequestResponder<ThreadsArgum
             responder.SetResponse(response);
         }
 
+        public static ProtocolMessages.Module ConvertToModule(in MODULE_INFO debugModuleInfo, IDebugModule2 module)
+        {
+            var mod = new ProtocolMessages.Module(RegisterDebugModule(module), debugModuleInfo.m_bstrName)
+            {
+                Path = debugModuleInfo.m_bstrUrl,
+                VsTimestampUTC = (debugModuleInfo.dwValidFields & enum_MODULE_INFO_FIELDS.MIF_TIMESTAMP) != 0 ? FileTimeToPosix(debugModuleInfo.m_TimeStamp).ToString(CultureInfo.InvariantCulture) : null,
+                Version = debugModuleInfo.m_bstrVersion,
+                VsLoadAddress = debugModuleInfo.m_addrLoadAddress.ToString(CultureInfo.InvariantCulture),
+                VsPreferredLoadAddress = debugModuleInfo.m_addrPreferredLoadAddress.ToString(CultureInfo.InvariantCulture),
+                VsModuleSize = (int)debugModuleInfo.m_dwSize,
+                VsLoadOrder = (int)debugModuleInfo.m_dwLoadOrder,
+                SymbolFilePath = debugModuleInfo.m_bstrUrlSymbolLocation,
+                SymbolStatus = debugModuleInfo.m_bstrDebugMessage,
+                VsIs64Bit = (debugModuleInfo.m_dwModuleFlags & enum_MODULE_FLAGS.MODULE_FLAG_64BIT) != 0
+            };
+            
+            // IsOptimized and IsUserCode are not set by gdb
+            if((debugModuleInfo.m_dwModuleFlags & enum_MODULE_FLAGS.MODULE_FLAG_OPTIMIZED) != 0)
+            {
+                mod.IsOptimized = true;
+            } else if ((debugModuleInfo.m_dwModuleFlags & enum_MODULE_FLAGS.MODULE_FLAG_UNOPTIMIZED) != 0)
+            {
+                mod.IsOptimized = false;
+            }
+            if (module is IDebugModule3 module3 && module3.IsUserCode(out int isUserCode) == HRConstants.S_OK)
+            {
+                if (isUserCode == 0) mod.IsUserCode = false;
```

Comment:
nit: not a fan of single line `if`/`else` without braces, causes problems if someone decides to add extra code underneath expecting it to run only in the `else` case.